profiles/permissions_test.py

```diff
@@ -187,3 +221,106 @@ def test_can_view_if_verified_mm_user(self):
+
+
+class CanSeeIfStaffTests(ESTestCase):
+    """
+    Tests for CanSeeIfStaff
+    """
+    def setUp(self):
+        super(CanSeeIfStaffTests, self).setUp()
+
+        with mute_signals(post_save):
+            self.instructor = UserFactory.create()
+            username = "{}_edx".format(self.instructor.username)
+            self.instructor.social_auth.create(
+                provider=EdxOrgOAuth2.name,
+                uid=username
+            )
+            ProfileFactory.create(
+                user=self.instructor,
+                verified_micromaster_user=False,
+            )
+            self.instructor_program = ProgramFactory.create()
+            Role.objects.create(
+                user=self.instructor,
+                role=Instructor.ROLE_ID,
+                program=self.instructor_program,
+            )
+
+            self.staff = UserFactory.create()
+            username = "{}_edx".format(self.staff.username)
+            self.staff.social_auth.create(
+                provider=EdxOrgOAuth2.name,
+                uid=username
+            )
+            ProfileFactory.create(
+                user=self.staff,
+                verified_micromaster_user=False,
+            )
+            self.staff_program = ProgramFactory.create()
+            Role.objects.create(
+                user=self.staff,
+                role=Staff.ROLE_ID,
+                program=self.staff_program,
+            )
+
+            # profile user starts off not assigned to any program
+            self.profile_user = profile_user = UserFactory.create()
+            username = "{}_edx".format(profile_user.username)
+            social_auth = profile_user.social_auth.create(
+                provider=EdxOrgOAuth2.name,
+                uid=username
+            )
+            self.profile_user_id = social_auth.uid
+            ProfileFactory.create(
+                user=profile_user,
+                verified_micromaster_user=False,
+                account_privacy=Profile.PRIVATE,
+            )
+
+    def test_staff_can_see_profile(self):
+        """
+        Staff can see private profile of user with same program
+        """
+        # Assign profile user to the staff user's program
+        ProgramEnrollment.objects.create(
+            program=self.staff_program,
+            user=self.profile_user,
+        )
+
+        perm = CanSeeIfNotPrivate()
+        staff_request = Mock(user=self.staff, is_anonymous=Mock(return_value=False))
+        instructor_request = Mock(user=self.instructor, is_anonymous=Mock(return_value=False))
+        view = Mock(kwargs={'user': self.profile_user_id})
+
+        ProgramEnrollment.objects.create(
+            user=self.profile_user,
+            program=ProgramFactory.create(),
+        )
+        assert perm.has_permission(staff_request, view) is True
+        with self.assertRaises(Http404):
+            perm.has_permission(instructor_request, view)
```

Comment:
You're right. Out of scope. I'll follow up with Giovanni later. 
> @pdpinch <https://github.com/pdpinch> 'instructor' and 'staff' are the two roles we created in #709 <https://github.com/mitodl/micromasters/pull/709>. We can change it but it would be out of the scope of this PR I think
>